### PR TITLE
rusk-wallet: Release prep UX changes

### LINKED ISSUE #642: wallet: UX enhancements
**Describe what you want implemented** 
- [x] Default to `key` 0 on all transaction commands in headless mode
- [x] Add defaults to `gas_limit` and `gas_price` in headless mode
- [x] Add `cargo install` to install the crate
- [x] Update `README.md` with binary name
- [x] Rephrase "offline" error as configuration error (not connection)
- [ ] Allow user to retry a wrong password
- [x] Have a minimum gas limit equal to the mempool minimum

**Describe "Why" this is needed**
Getting ready for distribution

### PR DESCRIPTION
**Pull request by t00ts**
Improvements:

- Default to key 0 on all transaction commands in headless mode
- Add defaults to gas_limit and gas_price in headless mode
- Add cargo install to install the crate
- Update README.md with binary name
- Rephrase "offline" error as configuration error (not connection)
- Have a minimum gas limit
- prevent address command from triggering a balance fetch

Resolves #642

### CODE REVIEW

File: rusk-wallet/src/lib/prompt.rs [hunk 1]
```diff
@@ -351,7 +351,7 @@ fn confirm(cmd: &CliCommand) -> bool {
             gas_limit,
             gas_price,
         } => {
-            let max_fee = gas_limit * gas_price.unwrap();
+            let max_fee = gas_limit.unwrap() * gas_price.unwrap();
```
Comment (vlopes11) [inline rusk-wallet/src/lib/prompt.rs:354]: Should `unwrap_or_default` for a constant `MIN_GAS_LIMIT · 1.07`

File: rusk-wallet/src/lib/prompt.rs [hunk 2]
```diff
@@ -380,7 +380,7 @@ fn confirm(cmd: &CliCommand) -> bool {
...
```
Comment (vlopes11) [inline rusk-wallet/src/lib/prompt.rs:383]: Ditto; unwrap for constant

File: rusk-wallet/src/lib/prompt.rs [hunk 3]
```diff
@@ -480,12 +480,12 @@ fn request_gas_limit() -> u64 {
     let question = requestty::Question::int("amt")
         .message("Introduce the gas limit for this transaction:")
         .default(DEFAULT_GAS_LIMIT as i64)
-        .validate_on_key(|n, _| n > 0)
+        .validate_on_key(|n, _| n > (MIN_GAS_LIMIT as i64))
```
Comment (vlopes11) [inline rusk-wallet/src/lib/prompt.rs:483]: The correct formula is `n > MIN_GAS_LIMIT + n * 0.07` (not taking into account the float implicit conversion)

File: rusk-wallet/src/lib/prompt.rs [hunk 4]
```diff
@@ -480,12 +480,12 @@ fn request_gas_limit() -> u64 {
...
         .validate(|n, _| {
-            if n > 0 {
-                Ok(())
+            if n < MIN_GAS_LIMIT as i64 {
```
Comment (vlopes11) [inline rusk-wallet/src/lib/prompt.rs:485]: Ditto; `n <= MIN_GAS_LIMIT + n * 0.07`

File: rusk-wallet/src/lib/wallet.rs [hunk 5]
```diff
@@ -288,6 +293,11 @@ impl CliWallet {
                     let my_addr = wallet.public_spend_key(key)?;
                     let mut rng = StdRng::from_entropy();
 
+                    let gas_limit = gas_limit.unwrap_or(DEFAULT_GAS_LIMIT);
+                    if gas_limit < MIN_GAS_LIMIT {
```
Comment (vlopes11) [inline rusk-wallet/src/lib/wallet.rs:297]: Ditto; `n <= MIN_GAS_LIMIT + n * 0.07`

**Review by vlopes11 — CHANGES_REQUESTED** [review]
Few nits

File: rusk-wallet/src/lib/prompt.rs (hunk 1 above)
Comment (t00ts) [reply rusk-wallet/src/lib/prompt.rs:354]: The option is **guaranteed** to have a `Some`

File: rusk-wallet/src/lib/prompt.rs [hunk 6]
```diff
@@ -368,7 +368,7 @@ fn confirm(cmd: &CliCommand) -> bool {
             gas_limit,
             gas_price,
         } => {
-            let max_fee = gas_limit * gas_price.unwrap();
+            let max_fee = gas_limit.unwrap() * gas_price.unwrap();
```
[the remark this answers is not in the document]
Comment (t00ts) [reply rusk-wallet/src/lib/prompt.rs:371]: The option is **guaranteed** to have a `Some`

File: rusk-wallet/src/lib/prompt.rs (hunk 3 above)
Comment (vlopes11) [reply rusk-wallet/src/lib/prompt.rs:489]: N/A

File: rusk-wallet/src/lib/prompt.rs (hunk 4 above)
Comment (vlopes11) [reply rusk-wallet/src/lib/prompt.rs:491]: N/A

File: rusk-wallet/src/lib/wallet.rs [hunk 7]
```diff
@@ -203,6 +198,11 @@ impl CliWallet {
                     let mut rng = StdRng::from_entropy();
                     let ref_id = BlsScalar::random(&mut rng);
 
+                    let gas_limit = gas_limit.unwrap_or(DEFAULT_GAS_LIMIT);
+                    if gas_limit < MIN_GAS_LIMIT {
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply rusk-wallet/src/lib/wallet.rs:202]: N/A

File: rusk-wallet/src/lib/wallet.rs [hunk 8]
```diff
@@ -237,6 +237,11 @@ impl CliWallet {
                     let my_addr = wallet.public_spend_key(key)?;
                     let mut rng = StdRng::from_entropy();
 
+                    let gas_limit = gas_limit.unwrap_or(DEFAULT_GAS_LIMIT);
+                    if gas_limit < MIN_GAS_LIMIT {
```
[the remark this answers is not in the document]
Comment (vlopes11) [reply rusk-wallet/src/lib/wallet.rs:241]: N/A

File: rusk-wallet/src/lib/wallet.rs (hunk 5 above)
Comment (vlopes11) [reply rusk-wallet/src/lib/wallet.rs:297]: N/A

**Review by vlopes11 — APPROVED** [review]
LGTM

**Pull request merged**